rfcs/0007-use-git2go.md

```diff
@@ -0,0 +1,69 @@
+## Problem:
+Git repositories with Azure Devops does not work with kpack.
+
+The issue stems from using the go-git library which does not support the git v2 protocol which is required with Azure Devops repos:
+- https://github.com/go-git/go-git/issues/64
+
+## Outcome
+Users can use Azure Devops for git repositories
+
+## Background
+
+Azure Devops represents a significant use case
+
+[go-git](https://github.com/go-git/go-git) is the git implementation library in pure go
+[git2go](https://github.com/libgit2/git2go) is a library for go bindings to the git method library in c, [libgit2](https://libgit2.org/).
+[Shallow cloning](https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---depthltdepthgt) basically means we don't pull in the full history which is a performance benefit.
```

Comment:
It is worth noting that we don't actually use shallow cloning at the [moment](https://github.com/pivotal/kpack/blob/12689fbfdf9d44a0d24d7acc435efcff8392f8c6/pkg/git/fetch.go#L51) (
depth: 0 is a full clone). My understanding is we do this to enable us to checkout commits that are not the tip of a branch.

Reply:
We could probably do a shallow clone if possible and I believe we discussed investigating that for performance issues but, haven't it seems because we haven't had that many performance concerns 🤷‍♂️.